jax/_src/lax/lax.py

```diff
@@ -5506,15 +5506,27 @@ def _operands_to_keys(*operands, num_keys=1):
 
 def _sort_jvp(primals, tangents, *, dimension, is_stable, num_keys):
   shape = primals[0].shape
-  iotas = []
-  for dim, size in enumerate(shape):
-    iotas.append(broadcasted_iota(np.int64, shape, dim))
   sorted_primals_and_idx = sort_p.bind(
-      *primals, iotas[dimension], dimension=dimension,
-      is_stable=is_stable, num_keys=num_keys)
-  idx = tuple(sorted_primals_and_idx[-1] if i == dimension else iotas[i]
-              for i in range(len(shape)))
-  tangents_out = tuple(t if type(t) is ad_util.Zero else t[idx] for t in tangents)
+      *primals, broadcasted_iota(np.uint64, shape, dimension),
+      dimension=dimension, is_stable=is_stable, num_keys=num_keys)
+  batch_dims = tuple(np.delete(np.arange(len(shape), dtype=np.int64),
+                               dimension))
+  dnums = slicing.GatherDimensionNumbers(
+    offset_dims=(),
+    collapsed_slice_dims=(dimension,),
+    start_index_map=(dimension,),
+    operand_batching_dims=batch_dims,
+    start_indices_batching_dims=batch_dims,
+  )
+  tangents_out = []
+  idx = expand_dims(sorted_primals_and_idx[-1], (len(shape),))
+  for t in tangents:
+    if type(t) is ad_util.Zero:
+      out = t
+    else:
+      out = slicing.gather(t, idx, dnums, slice_sizes=(1,) * len(shape),
+                           mode=slicing.GatherScatterMode.PROMISE_IN_BOUNDS)
+    tangents_out.append(out)
```

Comment:
Small suggestion: what do you think about writing it like this instead?
```python
gather_idx = partial(slicing.gather, start_indices=idx, dimension_numbers=dnums,
                     slice_sizes=(1,) * len(shape), mode=slicing.GatherScatterMode.PROMISE_IN_BOUNDS)
tangents_out = [t if type(t) is ad_util.Zero else gather_idx(t)]
```
I find short list comprehensions like this easier to read than appending within loops.

Reply:
Done.